### ISSUE TITLE
[Bug]: Screenshot only on failure takes all the chromium pages of running tests when run in parallel

### ISSUE THREAD

**Issue by changan1111** [microsoft/playwright#31789]
### Version

1.45.2

### Steps to reproduce

1) Created 3 Testcases and one intended for failure to capture screenshot.
2) it takes all the screenshot which is currently running when screenshot config is set for only on failure.

### Expected behavior

i would expect that failed page only

### Actual behavior

it takes all the screenshot which is currently running when screenshot config is set for only on failure.

### Additional context

Using Page object model.. 

I have created page/browser/context in the beforeEach and using that in the actual test 

### Environment

```shell
latest playwright 
OS: windows
```

**Comment by pavelfeldman** [comment]
I don't think this is related to parallel test runs - those run independently in different OS processes. It is more likely that your test somehow has several pages (error in manual page management?). In either case, please follow the issue template to file something we can reproduce.
